src/parser/mod.rs

```diff
@@ -0,0 +1,383 @@
+use crate::{
```

Comment:
the whole file seems quite readable, albeit a bit long - would it make sense to split into multiple files later maybe ?

Reply:
400 hundred lines seems ok to me. I mean, there's still work to do. It can be shorted or longer in the end and then we can split it. I don't like long files as well, just skipping this as it's still under the construction :)